### ISSUE TITLE
[Feature Request]: New Treegrid Carbon Component

### ISSUE THREAD

**Issue by Hayley-l-williams** [carbon-design-system/carbon#14116]
### The problem

Many product teams rely on a file directory pattern with the ability to perform action on the items in the directory. Currently there is no Carbon pattern to support this use case.

To design for this need, many teams use the Tree view Carbon component, however, this component is not built to support actions in the tree branches, therefore causing developers to break this component to meet the desired use case. Additionally, if they want to make this modified Tree view keyboard accessible, keyboard interactions cannot follow the interactions defined by Carbon (which is based off [w3's Tree view component](https://www.w3.org/WAI/ARIA/apg/patterns/treeview/)). 

 

### The solution

We hope Carbon can consider creating a component that supports taking actions on file directory items, similar to how [W3's Treegrid ](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/) operates. This will ensure developers do not have to continue customizing the Tree view and it can be keyboard accessible to meet accessibility compliance laws.

### Examples

Several "real-life" examples of teams using the Tree view with actions:
<img width="726" alt="Screenshot 2023-06-28 at 3 53 09 PM" src="https://github.com/carbon-design-system/carbon/assets/102319049/3480dcab-b0fb-42e9-9677-eeb52f4eadef">
<img width="981" alt="Screenshot 2023-06-28 at 3 53 38 PM" src="https://github.com/carbon-design-system/carbon/assets/102319049/6f6a4971-5738-4657-9dc9-38922ece983f">

This "tree grid" idea would be similar to how Outlook looks in that items can contain other items, and all items are actionable (Starring, flaggin, etc). I can't figure out how to access the actions with the keyboard however:
<img width="381" alt="Screenshot 2023-06-28 at 3 54 28 PM" src="https://github.com/carbon-design-system/carbon/assets/102319049/dc0eb40f-a57e-4fef-be02-48f7768f849a">

### Application/PAL

watsonx.data, Db2 console

### Business priority

Medium Priority = upcoming release but is not pressing

### Available extra resources

I'd be happy to help create this component; I could also see if a dev is interested in contributing as well

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by sstrubberg** [comment]
Need to figure out what other use-cases or teams would benefit. @Hayley-l-williams if you're interested in contributing this enhancement, flesh out the idea more and bring it to Design Office Hours. Thx!

**Comment by sstrubberg** [comment]
@Hayley-l-williams There is a separate call for Charts. @theiliad should be able to share the details of that call with you :D

**Comment by theiliad** [comment]
Yes @Hayley-l-williams pls reach out to me via DM, and we can get you on our call to discuss things
